api/v1alpha1/ratelimit_types.go

```diff
@@ -51,6 +51,16 @@ type GlobalRateLimit struct {
 	//
 	// +kubebuilder:validation:MaxItems=64
 	Rules []RateLimitRule `json:"rules"`
+
+	// Shared determines whether the rate limit rules apply across all the policy targets.
+	// If set to true, the rule is treated as a common bucket and is shared across all policy targets (xRoutes).
+	// Must have targetRef set to Gateway
+	// Default: false.
+	//
+	// +optional
+	// +notImplementedHide
+	// +kubebuilder:default=false
+	Shared *bool `json:"shared,omitempty"`
```

Comment:
Using `bool` is not recommended..
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md

Any strong reason for that? Can we consider enum instead?

Reply:
thanks for linking that @sanposhiho , it does say to think twice before using `bool`, mainly because intent/field could be multidimensional in the future (which enums like `type` really come in handy).
the intent here is to say that this ratelimit bucket is a singleton or is shared across all targets, can recommendations for enums here ?

Reply:
You can make a final decision based on other API designs in EG, but we've mostly (or probably always) used enum for new APIs even if we first thought it was a two-value switching and likely won't get changed in the future when we designed them. So, if you ask me, I'd probably change it to enum and change the name from `Shared`. (some general name, like `ApplyPolicy` etc)

Reply:
I like @rudrakhp's suggestion https://github.com/envoyproxy/gateway/pull/5227#discussion_r1957401918